src/Converter/LineItemsConverter.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\RefundPlugin\Converter;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Sylius\Component\Core\Model\AdjustmentInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Model\OrderItemUnitInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Sylius\RefundPlugin\Entity\LineItem;
+use Sylius\RefundPlugin\Entity\LineItemInterface;
+use Sylius\RefundPlugin\Model\UnitRefundInterface;
+use Webmozart\Assert\Assert;
+
+final class LineItemsConverter implements LineItemsConverterInterface
+{
+    /** @var RepositoryInterface */
+    private $orderItemUnitRepository;
+
+    public function __construct(RepositoryInterface $orderItemUnitRepository)
+    {
+        $this->orderItemUnitRepository = $orderItemUnitRepository;
+    }
+
+    public function convert(array $units): Collection
+    {
+        $lineItems = new ArrayCollection();
+
+        /** @var UnitRefundInterface $unitRefund */
+        foreach ($units as $unitRefund) {
+            /** @var OrderItemUnitInterface|null $orderItemUnit */
+            $orderItemUnit = $this->orderItemUnitRepository->find($unitRefund->id());
+            Assert::notNull($orderItemUnit);
+            Assert::lessThanEq($unitRefund->total(), $orderItemUnit->getTotal());
+
+            /** @var OrderItemInterface $orderItem */
+            $orderItem = $orderItemUnit->getOrderItem();
+
+            $grossValue = $unitRefund->total();
+            $taxAmount = (int) ($grossValue * $orderItemUnit->getTaxTotal() / $orderItemUnit->getTotal());
+            $netValue = $grossValue - $taxAmount;
```

Comment:
I feel that there is a lot of things we could extract there - even if they're done only there, it still does not seem nice 😢 Maybe we could have some VO like `CreditMemoValues` which would calculate these variables? Or some `LineItemFactory`? It's not a must-have, **for now,** for me, but would definitely improve a DX 🖖

Reply:
For now, I've extracted creating line item to the separate private method, but I agree that in future some refactor could improve a DX